app/code/Magento/Catalog/Model/Product.php

```diff
@@ -1729,6 +1729,8 @@ public function setIsDuplicable($value)
      * Check is product available for sale
      *
      * @return bool
+     * @deprecated
+     * @see \Magento\InventorySalesApi\Api\IsProductSalableInterface
```

Comment:
It might make sense to just link to devdocs to avoid confusing developers having no Inventory Installed.
https://devdocs.magento.com/guides/v2.3/inventory/inventory-api-reference.html